application/src/main/java/org/togetherjava/tjbot/CommandReloadUtil.java

```diff
@@ -0,0 +1,86 @@
+package org.togetherjava.tjbot;
+
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.requests.RestAction;
+import net.dv8tion.jda.api.requests.restaction.CommandListUpdateAction;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.togetherjava.tjbot.commands.BotCommand;
+import org.togetherjava.tjbot.commands.CommandVisibility;
+import org.togetherjava.tjbot.commands.system.CommandProvider;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+/**
+ * Offers utility functions for reloading all commands.
+ */
+public class CommandReloadUtil {
+    private static final Logger logger = LoggerFactory.getLogger(CommandReloadUtil.class);
+
+    private CommandReloadUtil() {
+        throw new IllegalStateException("Utility class");
```

Comment:
idiomatic exception is `UnsupportedOperationException` or whatever its called. or `AssertionError`.